API/Controllers/ProjectController.cs

```diff
@@ -792,6 +790,245 @@ public async Task<IActionResult> UnlikeProject(int projectId)
+                return NotFound(problemDetails);
+            }
+
+            bool isAllowed = await authorizationHelper.UserIsAllowed(currentUser,
+                                                                     nameof(Defaults.Scopes.AdminProjectWrite),
+                                                                     nameof(Defaults.Scopes.InstitutionProjectWrite),
+                                                                     project.UserId);
+
+            bool isCreatorAndHasScope =
+                project.IsCreator(currentUser.Id) &&
+                userService.UserHasScope(currentUser.IdentityId, nameof(Defaults.Scopes.ProjectWrite));
+
+            //If the current user isn't the creator of the project or the dataofficer of the organization which the creator belongs to
+            //Then this user is not authorized to link the institution to the project
+            if(!(isCreatorAndHasScope ||
+                 isAllowed))
+            {
+                return Forbid();
+            }
+
+            int projectCreatorInstitutionId = project.User.InstitutionId.GetValueOrDefault();
+
+            if(projectCreatorInstitutionId == default)
+            {
+                ProblemDetails problemDetails = new ProblemDetails
+                {
+                    Title = "Failed setting the status of the project.",
+                    Detail = "The creator of the project send in the request is not bound to an institution and can therefore not make the project private.",
+                    Instance = "6972f184-7715-41e6-8bca-8b4ee63d5c58"
+                };
+                return NotFound(problemDetails);
+            }
+
+            //Link institution of the creator of the project to project if the institution isn't linked to the project yet
+            if(institutePrivate &&
+                !projectInstitutionService.InstitutionIsLinkedToProject(projectId, projectCreatorInstitutionId))
+            {
+                ProjectInstitution projectInstitution = new ProjectInstitution(projectId, projectCreatorInstitutionId);
+                await projectInstitutionService.AddAsync(projectInstitution).ConfigureAwait(false);
+            }
+
+            project.InstitutePrivate = institutePrivate;
+            projectService.Update(project);
+
+            projectService.Save();
+
+            return Ok(mapper.Map<Project, ProjectResourceResult>(project));
+        }
+
+        /// <summary>
+        /// Links given project and institution. This function is admin only!
+        /// </summary>
+        /// <param name="projectId">The project identifier</param>
+        /// <param name="institutionId">The institution identifier</param>
+        /// <returns>
+        ///     StatusCode 201 If success,
+        ///     StatusCode 404 If the project, institution or current user couldn't be found.
+        ///     StatusCode 409 If the project is already linked to the institution.
+        /// </returns>
```

Comment:
Try to use these response comments because Swagger uses it for documentation.
https://github.com/DigitalExcellence/dex-backend/blob/develop/API/Controllers/CallToActionOptionController.cs#L89